linux/src/async_socket_linux.c

```diff
@@ -128,8 +140,21 @@ static int on_socket_recv(void* context, ASYNC_SOCKET_HANDLE async_socket, void*
     }
     else
     {
-        // Codes_SRS_ASYNC_SOCKET_LINUX_04_007: [ on_socket_recv shall attempt to receive data by calling the system recv socket API. ]
-        result = recv(async_socket->socket_handle, buf, len, 0);
+        // Codes_SRS_ASYNC_SOCKET_LINUX_04_007: [ on_socket_recv shall attempt to receive data by calling the socket_transport_receive API. ]
+        SOCKET_BUFFER input_buf;
+        input_buf.buffer = buf;
+        input_buf.length = len;
+        uint32_t bytes_recv;
+        if(socket_transport_receive(async_socket->socket_transport_handle, &input_buf, 1, &bytes_recv, 0, NULL) != SOCKET_RECEIVE_OK)
```

Comment:
>if(socket_transport_receive(async_socket->socket_transport_handle, &input_buf, 1, &bytes_recv, 0, NULL) != SOCKET_RECEIVE_OK) [](http://example.com/codeflow?start=8&length=125)

There is nuance here that you need to take into account.  Let's talk offline because it's too much to describe here.